Mathlib/FieldTheory/JacobsonNoether.lean

```diff
@@ -0,0 +1,215 @@
+/-
+Copyright (c) 2024 F. Nuccio, H. Zheng, W. He, S. Wu, Y. Yuan, W. Jiao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Filippo A. E. Nuccio, Huanyu Zheng, Sihan Wu, Wanyi He, Weichen Jiao, Yi Yuan
+-/
+import Mathlib.Algebra.CharP.LinearMaps
+import Mathlib.Algebra.CharP.Subring
+import Mathlib.Algebra.Field.Power
+import Mathlib.FieldTheory.PurelyInseparable
+
+/-!
+# The Jacobson-Noether theorem
+
+This file contains a proof of the Jacobson-Noether theorem and some auxiliary lemmas.
+Here we discuss different cases of characteristics of
+the noncommutative division algebra `D` with center `k`.
+
+## Main Results
+
+- `exists_separable_mem_of_not_central` : (Jacobson-Noether theorem) For a
+  non-commutative algebraic division algebra `D` (with base ring
+  being its center `k`), then there exist an element `x` of
+  `D \ k` that is separable over its center.
+- `exists_separable_mem_of_not_central'` : (Jacobson-Noether theorem) For a
+  non-commutative algebraic division algebra `D` (with base ring
+  being a field `L`), if the center of `D` over `L` is `L`,
+  then there exist an element `x` of `D \ L` that is separable over `L`.
+
+## Notations
+
+- `D` is a noncommutative division algebra
+- `k` is the center of `D`
+
+## Implementation Notes
+
+Mathematically, `exists_separable_mem_of_not_central` and `exists_separable_mem_of_not_central'`
+are equivalent.
+
+The difference however, is that the former takes `D` as the only variable
+and fixing `D` would forces `k`. Whereas the later takes `D` and `L` as
+separate variables constrained by certain relations.
+
+## Reference
+* <https://ysharifi.wordpress.com/2011/09/30/the-jacobson-noether-theorem/>
+-/
+
+namespace JacobsonNoether
+
+variable {D : Type*} [DivisionRing D]
+
+open LinearMap
+
+local notation3 "k" => (Subring.center D)
```

Comment:
I'm not entirely sure about this. I would think the mathliby approach is to have `[Algebra k D]` and a new typeclass called `IsCentral k D`. 

But if you have managed to restate all the results under that generality then it's fine.

Reply:
Good point, I thought you might mean `exists_separable_mem_of_not_central'`, but `Algebra.IsCentral` seems more general than the `hcenter` in it... I'll try to generalize that a bit. Thanks!

Reply:
Thanks! I've switched `exists_separable_mem_of_not_central'` to `IsCentral`.